README.md

```diff
@@ -1 +1,160 @@
+[kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) tool. Here are
+some examples:
+
+- `kubectl get pods`: lists all pods.
+- `kubectl get svc`: lists all services. This will list which ports are open on
+  all pods.
+- `kubectl get ing`: lists all ingress resources. Ingress resources are external
+  communication interfaces. In this particular system there's currently only one
+  ingress, getting to port 80 in the `nginx` container.
+
+To get details from running pods you can use the `kubectl logs` command.
+ex.
+```
+kubectl logs cytomine-core-6db5645df7-pcd9z
+```
+(use `kubectl logs -f` to get auto-updating logs). Note that you need to use the
+full pod name including the id parts to get logs.
+
+To get the state of a pod from kuberentes standpoint, you can use `describe`.
+ex. `kubectl describe deployment cytomine-core` instead. That is particularly
+useful to get information on why a pod isn't starting, as you can't get logs
+from pods that aren't running.
+
+### Helm
+
+[Helm](https://helm.sh/docs/intro/install) is a package manager for kubernetes,
+which allows us to use templates which will be automatically filled with values.
+This makes management much more convenient.
+
+## Development environment
+
+This development environment has been tested in minikube. Instructions for
+installing minikube can be found in the
+[official instructions](https://minikube.sigs.k8s.io/docs/start/).
+
+Parts of the internal message authentication in cytomine requires that the web
+client runs on a url that's reachable from other containers, making it difficult
+to run on localhost. Because of this you will also need to follow the
+instructions for using the
+[ingress dns](https://minikube.sigs.k8s.io/docs/handbook/addons/ingress-dns)
+add-on.
+
+### Setting up the environment on OS X
+
+Minikube doesn't seem to be superhappy on OS X, but with a few precautions it
+works!
+
+-  If you wish to use the `ingress-dns` addon (which is used in this project),
+   you need to use the `hyperkit` driver (start minikube with:
+   `minikube start --driver=hyperkit`). I tried using the docker driver but
+   couldn't get it to work on my machine.
+
+-  Connecting the cisco VPN client seems to mess with minikubes networking. I
+   need to restart minikube every time I need to use the VPN.
+
+-  Docker desktop also seems to conflict with minikube, so I can't run both
+   docker and minikube at the same time (unless I use the docker driver, but
+   then ingress-dns doesn't work).
+
+### Setting up the environment on Ubuntu System
+
+We need to add the `minikube-ip` as a DNS server
+
+Update the file `/etc/resolvconf/resolv.conf.d/base` to have the following contents
+
+```
+search test
+nameserver <minikube ip>
+timeout 5
+```
+
+Ensure your Linux OS uses `systemcl`, run the following command
+
+`sudo resolvconf -u`
+
+To verify, run the following command to check if the `/etc/resolv.conf` is updated with the minikube ip.
+

```

Comment:
Strangely, using `resolvconf` in Ubuntu for adding the DNS did not work for me, but the following quick fix did:
````suggestion

If the above steps do not work or you do not want to use `resolvconf`, add the above contents manually instead. *After** `minikube` is started run
```
echo -e $"\nsearch test\nnameserver $(minikube ip)\ntimeout 5" \
| sudo tee -a /etc/resolv.conf
```
When you are done testing cytomine, you can cleanup `resolv.conf` or wait till this is done automatically when the dhcp lease is renewed.

````